libs/positions/src/lib/use-positions-data.tsx

```diff
@@ -37,35 +38,43 @@ const getSummaryRowData = (positions: Position[]) => {
 
 export const usePositionsData = (
   partyId: string,
-  gridRef: RefObject<AgGridReact>,
-  assetSymbol?: string
+  gridRef: RefObject<AgGridReact>
 ) => {
-  const variables = useMemo(() => ({ partyId }), [partyId]);
+  const variables = useMemo<PositionsMetricsProviderVariables>(
+    () => ({ partyId }),
+    [partyId]
+  );
   const dataRef = useRef<Position[] | null>(null);
   const update = useCallback(
-    ({ data }: { data: Position[] | null }) => {
-      dataRef.current = assetSymbol ? filter(data, { assetSymbol }) : data;
-      gridRef.current?.api?.refreshInfiniteCache();
-      return true;
-    },
-    [assetSymbol, gridRef]
-  );
-  const { data, error, loading } = useDataProvider<Position[], never>({
-    dataProvider,
-    update,
-    variables,
-  });
-  if (!dataRef.current && data) {
-    dataRef.current = assetSymbol ? filter(data, { assetSymbol }) : data;
-  }
-  const getRows = useCallback(
-    async ({ successCallback, startRow, endRow }: GetRowsParams) => {
-      const rowsThisBlock = dataRef.current
-        ? dataRef.current.slice(startRow, endRow)
-        : [];
-      const lastRow = dataRef.current?.length ?? -1;
-      successCallback(rowsThisBlock, lastRow);
-      if (gridRef.current?.api) {
+    ({
+      data,
+      delta,
+    }: {
+      data: Position[] | null;
+      delta?: Position[] | null;
+    }) => {
+      dataRef.current = data;
+
+      const update: Position[] = [];
+      const add: Position[] = [];
```

Comment:
question (blocking): Did you intend to push new rows to the `add`,`remove` arrays?

Reply:
Good catch. I forgot to handle adding missing row